docs/architecture/adr-007-specialization-groups.md

```diff
@@ -0,0 +1,178 @@
+some of these possiblities in a common interface dubbed the `Electionator`. For
+its initial implementation as a part of this ADR we recommend that the general
+election abstraction (`Electionator`) is provided as well as a basic
+implementation of that abstraction which allows for a continuous election of
+members of a specialization group. 
+
+``` golang
+// The Electionator abstraction covers the concept space for 
+// a wide variety of election kinds.  
+type Electionator interface {
+    
+    // is the election object accepting votes.
+    Active() bool 
+
+    // functionality to execute for when a vote is cast in this election, here
+    // the vote field is anticipated to be marshalled into a vote type used 
+    // by an election. 
+    // 
+    // NOTE There are no explicit ids here. Just votes which pertain specifically
+    // to one electionator. Anyone can create and send a vote to the electionator item
+    // which will presumably attempt to marshal those bytes into a particular struct
+    // and apply the vote information in some arbitrary way. There can be multiple
+    // Electionators within the Cosmos-Hub for multiple specialization groups, votes
+    // would need to be routed to the Electionator upstream of here.
+    Vote(addr sdk.AccAddress, vote []byte) 
+
+    // here lies all functionality to authenticate and execute changes for
+    // when a member accepts being elected
+    AcceptElection(sdk.AccAddress) 
+
+    // Register a revoker object
+    RegisterRevoker(Revoker)
+
+    // No more revokers may be registered after this function is called
+    SealRevokers()
+
+    // register hooks to call when an election actions occur
+    RegisterHooks(ElectionatorHooks) 
+
+    // query for the current winner(s) of this election based on arbitrary
+    // election ruleset
+    QueryWinners() []sdk.AccAddress 
+
+    // query metadata for an address in the election this 
+    // could include for example position that an address
+    // is being elected for within a group   
+    // 
+    // this metadata may be directly related to 
+    // voting information and/or privileges enabled
+    // to members within a group. 
+    QueryMetadata(sdk.AccAddress) []byte
+}
+
+// ElectionatorHooks, once registered with an Electionator, 
+// trigger execution of relevant interface functions when 
+// Electionator events occur. 
+type ElectionatorHooks interface {
+    AfterVoteCast(addr sdk.AccAddress, vote []byte)
+    AfterMemberAccepted(addr sdk.AccAddress)
+    AfterMemberRevoked(addr sdk.AccAddress, cause []byte)
+}
+
+// Revoker defines the function required for an membership revocation rule-set
+// used by a specialization group. This could be used to create self revoking,
+// and evidence based revoking, etc. Revokers types may be created and
+// reused for different election types. 
+// 
+// When revoking the "cause" bytes may be arbitrarily marshalled into evidence,
+// memos, etc.
+type Revoker interface {
```

Comment:
There should be a way for governance to disband an entire specialization group if they are caught colluding maliciously

Reply:
added:  `emergency disband of the entire dCERT group (ex. for colluding maliciously)` to ADR 008 - I think this is not a generalized specialization group problem but relevant for dCERT

Reply:
Also added this 
```
Additionally in the case of an emergency situation of a colluding and malicious
dCERT group, the community needs the capability to disband the entire dCERT
group and likely fully slash them. This could be achieved though a special new
proposal type (general governance) which would halt the functionality of the
dCERT group until the proposal was concluded. This special proposal type would
likely need to also have a fairly large wager which could be slashed if the
proposal creator was malicious. 
```